src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php

```diff
@@ -344,6 +345,32 @@ public function testDumpOverridenGetters()
         $this->assertSame('baz', $r->invoke($baz));
     }
 
+    public function testDumpOverridenGettersWithConstructor()
+    {
+        $container = include self::$fixturesPath.'/containers/container_dump_overriden_getters_with_constructor.php';
+        $container->compile();
+        $container->getDefinition('foo')
+            ->setOverriddenGetter('getInvalid', array(new Reference('bar', ContainerBuilder::IGNORE_ON_INVALID_REFERENCE)));
+        $dumper = new PhpDumper($container);
+
+        $dump = $dumper->dump(array('class' => 'Symfony_DI_PhpDumper_Test_Overriden_Getters_With_Constructor'));
+        $this->assertStringEqualsFile(self::$fixturesPath.'/php/services_dump_overriden_getters_with_constructor.php', $dump);
+        $resources = array_map(
```

Comment:
@nicolas-grekas we actually need to check for a file resource, as you did in #21505